datalad/distributed/create_sibling_ghlike.py

```diff
@@ -0,0 +1,549 @@
+        raise RuntimeError(f'Unexpected host response: {response}')
+
+
+def _response2repoprops(response):
+    """
+    """
+    return dict(
+        reponame=response.get('name'),
+        private=response.get('private'),
+        clone_url=response.get('clone_url'),
+        ssh_url=response.get('ssh_url'),
+        html_url=response.get('html_url'),
+    )
+
+
+def _create_sibling(
+        platform,
+        reponame,
+        dataset=None,
+        recursive=False,
+        recursion_limit=None,
+        name=None,
+        existing='error',
+        access_protocol='https',
+        publish_depends=None,
+        private=False,
+        dry_run=False):
+    """
+    """
+
+    orgname, reponame = split_org_repo(reponame)
+
+    # apply whatever normalization or default selection
+    name = platform.get_siblingname(name)
+
+    lgr.debug("Repository organization name: '%s'", orgname)
+
+    if reponame != platform.normalize_reponame(reponame):
+        raise ValueError(
+            f'Invalid name for a {platform.fullname} project: {reponame}')
+
+    lgr.debug("Repository basename: '%s'", reponame)
+
+    # what to operate on
+    ds = require_dataset(
+        dataset,
+        check_installed=True,
+        purpose=f'create {platform.fullname} sibling(s)')
+
+    res_kwargs = dict(
+        action=\
+        f'create_sibling_{platform.name} [dry-run]'
+        if dry_run else
+        f'create_sibling_{platform.name}',
+        logger=lgr,
+        refds=ds.path,
+    )
+
+    toprocess, filterresults = platform.get_dataset_reponame_mapping(
+        ds, name, reponame, existing, recursive, recursion_limit,
+        res_kwargs
+    )
+    yield from filterresults
+
+    if not toprocess:
+        # all skipped
+        return
+
+    lgr.debug("Will process %i dataset(s)", len(toprocess))
+
+    yield from platform.create_repos(
+        toprocess,
+        name,
+        orgname,
+        private,
+        dry_run,
+        res_kwargs,
+        existing,
+        access_protocol,
+        publish_depends,
+    )
+
+
+def split_org_repo(name):
+    """Split a potential organization name prefix from a repo's full name
+
+    Returns
+    -------
+    (None, reponame) or (orgname, reponame)
+    """
+    split = name.split('/', maxsplit=1)
+    if len(split) < 2:
+        return None, name
+    else:
+        return split[0], split[1]
+
+
+def _get_present_datasets(ds, recursive, recursion_limit):
+    """
+    """
+    # gather datasets and essential info
+    # dataset instance and mountpoint relative to the top
+    toprocess = [ds]
+    if recursive:
+        for sub in ds.subdatasets(
+                # we want to report on missing dataset in here
+                fulfilled=None,
+                recursive=recursive,
+                recursion_limit=recursion_limit,
+                result_xfm='datasets'):
+            if not sub.is_installed():
+                lgr.info('Ignoring unavailable subdataset %s', sub)
```

Comment:
I thought that using logger (INFO level in particular) for such purposes is not in fashion any longer... should it return annotated dict of what to process and what to "skip" for some reason, so code above could yield corresponding result records?

Reply:
Good point. This is a C&P from github code.